packages/core/src/external/carequality/dq/process-incoming-dq.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  DocumentQueryReqFromExternalGW,
+  DocumentQueryRespToExternalGW,
+} from "@metriport/ihe-gateway-sdk";
+import {
+  XDSUnknownPatientId,
+  XDSUnknownCommunity,
+  XDSMissingHomeCommunityId,
+  XDSRegistryError,
+  validateDQ,
+} from "./validating-dq";
+import { CODE_SYSTEM_REQUIRED_ERROR as DQ_CODE_SYSTEM_REQUIRED_ERROR } from "../shared";
+
+function constructErrorResponse(
+  payload: DocumentQueryReqFromExternalGW,
+  codingSystem: string,
+  code: string,
+  error: string
+): DocumentQueryRespToExternalGW {
+  return {
+    id: payload.id,
+    timestamp: payload.timestamp,
+    responseTimestamp: new Date().toISOString(),
+    operationOutcome: {
+      resourceType: "OperationOutcome",
+      id: payload.id,
+      issue: [
+        {
+          severity: "error",
+          code: "processing",
+          details: {
+            coding: [{ system: codingSystem, code: code }],
+            text: error,
+          },
+        },
+      ],
+    },
+  };
+}
+
+export async function processIncomingRequest(
+  payload: DocumentQueryReqFromExternalGW
+): Promise<DocumentQueryRespToExternalGW> {
+  try {
+    // validate incoming request + look for patient and get all their documents from s3
+    const documentContents = await validateDQ(payload);
+
+    // construct response
+    const response: DocumentQueryRespToExternalGW = {
+      id: payload.id,
+      timestamp: payload.timestamp,
+      responseTimestamp: new Date().toISOString(),
+      extrinsicObjectXmls: documentContents,
+    };
+
+    return response;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    switch (error.constructor) {
+      case XDSUnknownPatientId:
+        return constructErrorResponse(
```

Comment:
nit: For each switch case, there's 3/4 lines that are identical. Perhaps refactor to avoid duplication

Reply:
The downstream one has all the error code refactored :)
https://github.com/metriport/metriport/pull/1405